File: frontend/src/types/summary.ts
// frontend/src/types/summary.ts
export interface MonthlyExpenseSummaryItem {
  month_name: string; // e.g., "January", "February" (matching backend schema)
  year: number;
  total_expenses: number;
}

export interface DailyExpenseSummaryItem {
  date_str: string; // YYYY-MM-DD (matching backend schema)
  day_name: string; // e.g., "Monday", "Tuesday" (matching backend schema)
  total_expenses: number;
}

// This will be the structure recharts expects after transformation
export interface ChartDataItem {
  name: string; // Mapped from month_name or day_name/date_str
  expenses: number; // Mapped from total_expenses
}

// This is the overall response structure from the backend API
export interface UserExpenseSummaryResponse {
  monthly_summary: MonthlyExpenseSummaryItem[];
  daily_summary: DailyExpenseSummaryItem[];
}

File: backend/app/schemas/summary.py
# backend/app/schemas/summary.py
from pydantic import BaseModel
from typing import List
import datetime

class BaseSummaryItem(BaseModel):
    total_expenses: float

class MonthlyExpenseItem(BaseSummaryItem):
    month: int # 1 for January, 2 for February, etc.
    month_name: str # e.g., "January", "February"
    year: int

class DailyExpenseItem(BaseSummaryItem):
    date_str: str # YYYY-MM-DD
    day_name: str # e.g., "Monday", "Tuesday"

class UserExpenseSummaryResponse(BaseModel):
    monthly_summary: List[MonthlyExpenseItem]
    daily_summary: List[DailyExpenseItem] # For weekly, if we decide to do daily for past 7 days
